addons/knowledge/static/src/js/tools/tree_panel_mixin.js

```diff
@@ -28,13 +28,7 @@ export default {
         }).then(htmlTree => {
             $container.empty();
             $container.append(htmlTree);
-            if (!portalReadonlyMode) {
-                this._setTreeListener();
-                this._renderEmojiPicker();
-            }
-            this._setTreeFavoriteListener();
-        }).catch(error => {
-            $container.empty();
+            return $container;
```

Comment:
shouldn't we await the for rpc then return the container ? (so that the return thing of _renderTree is the container and not the RPC that return the container.
I'm not super confortable with async brols so maybe I'm à côté de la plaque